File: Java/RecyclerView10RecyclerViewWithOnClickListener/app/src/main/java/dev/ogabek/recyclerview10_recyclerviewwithonclicklistener/activity/MainActivity.java
package dev.ogabek.recyclerview10_recyclerviewwithonclicklistener.activity;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;
import dev.ogabek.recyclerview10_recyclerviewwithonclicklistener.R;
import dev.ogabek.recyclerview10_recyclerviewwithonclicklistener.adapter.MainRecyclerViewAdapter;
import dev.ogabek.recyclerview10_recyclerviewwithonclicklistener.model.Member;

public class MainActivity extends AppCompatActivity {

    Context context;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();

        refreshAdapter(prepareMembersList());

    }

    private List<Member> prepareMembersList() {
        List<Member> members = new ArrayList<>();
        for (int i = 0; i < 25; i++) {
            members.add(new Member(i + 1, "Ogabek Matyakubov", "+998 99 749 25 81"));
        }
        return members;
    }

    private void initView() {
        context = this;
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context, 1));
    }

    private void refreshAdapter(List<Member> members) {
        MainRecyclerViewAdapter adapter = new MainRecyclerViewAdapter(this, members);
        recyclerView.setAdapter(adapter);
    }

    public void ShowDialog(String title, String message) {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle(title);
        builder.setMessage(message);
        builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        builder.create().show();
    }

}

File: Kotlin/RecyclerView10RecyclerViewWithOnClickListener/app/src/main/java/dev/ogabek/recyclerview_10_recyclerviewwithonclicklistener/activity/MainActivity.kt
package dev.ogabek.recyclerview_10_recyclerviewwithonclicklistener.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview_10_recyclerviewwithonclicklistener.R
import dev.ogabek.recyclerview_10_recyclerviewwithonclicklistener.adapter.MainAdapter
import dev.ogabek.recyclerview_10_recyclerviewwithonclicklistener.model.Member

class MainActivity : AppCompatActivity() {
    private lateinit var context: MainActivity
    private lateinit var recyclerView: RecyclerView
    private lateinit var adapter: MainAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        refreshAdapter(prepareMemberList())

    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
        recyclerView.isNestedScrollingEnabled = false
    }

    private fun refreshAdapter(members: List<Member>) {
        adapter = MainAdapter(context, members)
        recyclerView.adapter = adapter
    }

    private fun prepareMemberList(): List<Member> {
        val members = ArrayList<Member>()
        for (i in 0..50) {
            members.add(Member(R.drawable.img_user, "Ogabek Matyakubov", "ID Number $i"))
        }
        return members
    }

    public fun showDialogMessage(title: String, message: String) {
        val dialog = AlertDialog.Builder(context)
        dialog.apply {
            setTitle(title)
            setMessage(message)
            setPositiveButton("OK") { dialog, _ ->
                dialog.dismiss()
            }
            show()
        }
    }

}